### PR TITLE
Continue form type optimization

### PR DESCRIPTION
**Pull request by FabienPapet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Refactor form types service definitions
| Type?             | improvement
| Category?         | CO
| BC breaks?        | yes
| Deprecations?     |  no
| How to test ?     |  This is a major refactoring on how forms are generated. It should have no impact at all, but it requires to go to every form in the application to check. A lot of checks can be done with automated tests, as we only check if the form is displayed. If the form is displayed, it will be working as it used to. The acceptance here is just the form to be successfully displayed.

### CODE REVIEW

File: src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml [hunk 1]
```diff
@@ -3,39 +3,58 @@ services:
...
-    # Add the console.command tag to all services defined in this file having this type
-    PrestaShopBundle\Form\Admin\Type\TranslatorAwareType:
-      bind:
-        $locales: "@=service('prestashop.adapter.legacy.context').getLanguages()"
+    bind:
+      # We need to get rid of these service calls, but at least, they now all have the same name.
+      # Symfony will throw an error when an argument is no more used in all services definitions so we can safely remove them.
+      # Before adding your own, please check it does not already exist in the list.
+      # There could be some duplicates.
+      $locales: "@=service('prestashop.adapter.legacy.context').getLanguages()"
```
Comment (matthieu-rolland) [inline src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml:11]: nice 👍

**Review by matthieu-rolland — DISMISSED** [review]
lgtm

**Review by mflasquin — DISMISSED** [review]

**Comment by AureRita** [comment]
Hello @FabienPapet 

Thank you for your PR ! I tested it and I find this issue on it : 

You can create a link list as you can see : 

[Untitled_ Feb 28, 2023 11_59 AM.webm](https://user-images.githubusercontent.com/100699445/221870104-abe9a7c3-e2b2-42e3-a1d9-d3d15762b3e7.webm)

**Review by matthieu-rolland — APPROVED** [review]

**Review by mflasquin — APPROVED** [review]

**Review by MhiriFaten — APPROVED** [review]
Hello @FabienPapet ,

I have checked your PR and LGTM.
Automated tests are OK !

QA approved :heavy_check_mark: 

Thank you.

**Comment by nicosomb** [comment]
Thank you @FabienPapet and @MhiriFaten !

**Pull request merged**